sdk/v2/core/logs.go

```diff
@@ -0,0 +1,152 @@
+package core
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/brigadecore/brigade/sdk/v2/internal/restmachinery"
+)
+
+// LogsSelector represents useful criteria for selecting logs for streaming from
+// a specific container of a Worker or Job.
+type LogsSelector struct {
+	// Job specifies, by name, a Job spawned by the Worker. If this field is
+	// left blank, it is presumed logs are desired for the Worker itself.
```

Comment:
Could the `LogsSelector` object have a field for explicitly retrieving the worker logs, rather than implicitly doing so when `Job` is empty?

Reply:
I had thought about that as well. There's a 1:1 between events and workers, so when you pass the eventID when retrieving logs, you have implicitly identified a worker. So it's not so much that the _absence_ of a value in the `job` field == a request for worker logs. It's more that `job` is an optional field for drilling down past what you've already asked for.

The function signature for getting logs might make that more apparent:

```
func (l *logsClient) Stream(
	ctx context.Context,
	eventID string,
	selector LogsSelector,
	opts LogStreamOptions,
) (<-chan LogEntry, <-chan error, error)
```

Thoughts? Maybe the comments on the `Job` field just need to be worded a bit better?

Reply:
Yeah, a comment describing this behavior would definitely help.

Reply:
Fix incoming shortly.

Reply:
@radu-matei please lmk if db586274cce1fb9f5beb07f37111d5d1eafe1413 clears things up.